pkg/kubelet/kubelet_pods.go

```diff
@@ -139,6 +139,15 @@ func (kl *Kubelet) makeBlockVolumes(pod *v1.Pod, container *v1.Container, podVol
 	return devices, nil
 }
 
+// shouldMountWindowsHostFile checks if the host file needs to be mapped when running on windows
+// if Windows is a hostProcess then the pod will use the nodes host file without mounting
+func shouldMountWindowsHostFile(pod *v1.Pod) bool {
+	if runtime.GOOS != "windows" {
+		return true
+	} 
+	return !*pod.Spec.SecurityContext.WindowsOptions.HostProcess
```

Comment:
I'd check if it's set on the container itself, we were thinking about for beta+ how we might be able to run server containers alongside host process, so a pod wide check would work currently but would need to be adjusted in the future.

Reply:
modified to check all containers on the pod